scipy/optimize/_qap.py

```diff
@@ -0,0 +1,780 @@
+        less than the number of nodes.
+
+    init_J : 2d-array or "barycenter" (default = "barycenter")
+        The initial (guess) permutation matrix or search "position"
+        :math:`J`.
+
+        :math:`J` need not be a proper permutation matrix;
+        however, it must have the same shape as `A` and
+        `B`, and it must be doubly stochastic: each of its
+        rows and columns must sum to 1.
+
+        If unspecified or ``"barycenter"``, the non-informative "flat
+        doubly stochastic matrix" :math:`1*1^T/n`, where :math:`n`
+        is the number of nodes and :math:`1` is a :math:`n \times 1`
+        array of ones, is used. This is the "barycenter" of the
+        search space of doubly-stochastic matrices.
+
+    init_weight : float in range [0, 1]
+        Allows the user to specify the weight of the provided
+        search position :math:`J` relative to random perturbations
+        :math:`K`.
+
+        The algorithm will be repeated :math:`k` times
+        from randomized initial search positions
+        :math:`P_0 = (\alpha J + (1- \alpha) K`, where
+        :math:`J` is given by option `init_J`,
+        :math:`\alpha` is given by option `init_weight`,
+        :math:`k` is given by option `init_k`, and
+        :math:`K` is a random doubly stochastic matrix.
+
+        Default is 1 if `init_k` is 1, 0 otherwise.
+    init_k : int, positive (default = 1)
+        The number of randomized initial search positions :math:`P_0`
+        from which the FAQ algorithm will proceed.
+    maxiter : int, positive (default = 30)
+        Integer specifying the max number of Franke-Wolfe iterations
+        per initial search position. FAQ typically converges within a
+        modest number of iterations.
+    shuffle_input : bool (default = True)
+        To avoid artificially high or low matching due to inherent
+        sorting of input matrices, gives users the option
+        to shuffle the nodes. Results are then unshuffled so that the
+        returned results correspond with the node order of inputs.
+    eps : float (default = 0.05)
+        A threshold for the stopping criterion. Franke-Wolfe
+        iteration terminates when the change in search position between
+        iterations is sufficiently small, that is, when the Frobenius
+        norm of :math:`(P_{i}-P_{i+1}) \leq eps`, where :math:`i` is
+        the iteration number.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` containing the following
+        fields.
+
+        col_ind : 1-D array
+            An array of column indices corresponding with the best
+            permutation of the nodes of `B` found.
+        score : float
+            The corresponding value of the objective function.
+        nit : int
+            The number of Franke-Wolfe iterations performed during
+            the initialization resulting in the permutation
+            returned.
```

Comment:
This is very unclear. How about `The number of Franke-Wolfe iterations performed.`?

Reply:
That wasn't what it was doing. But I removed multistart.

Reply:
I still prefer a shorter sentence.
What part is the initialization? The optimization while loop?

Reply:
Oops. I changed it in `maxiter` not here. Fixed.